src/modules/mavlink/mavlink_main.cpp

```diff
@@ -1782,6 +1790,12 @@ Mavlink::task_main(int argc, char *argv[])
 			set_protocol(SERIAL);
 			break;
 
+		case 'n':
+#ifdef __PX4_POSIX_BBBLUE
+			_mavlink_wifi_name = myoptarg;
```

Comment:
Why do we need a special case for wifi? Why is it any different than a regular network interface?

Reply:
In posix-configs/bbblue/px4.config, I have the following comments about the usage of -n option.

mavlink start -x -u 14556 -r 1000000 
   -n name of wifi interface: SoftAp, wlan or your custom interface, 
   e.g., -n wlan . The default on BBBlue is SoftAp

Besides the regular network interface, a SoftAP is also activated after boot by default on BBBlue.
So there are 2 not 1 active wifi interfaces. The existing code only picks the regular one.